app/workers/digest_email_generation_worker.rb

```diff
@@ -50,4 +55,8 @@ def subscriber_content_changes
       digest_run: digest_run
     )
   end
+
+  def update_digest_run
+    digest_run.mark_complete! unless DigestRunSubscriber.incomplete_for_run(digest_run.id).exists?
```

Comment:
I might make `DigestRunSubscriber.incomplete_for_run(digest_run.id).exists?` into a method on the `DigestRun` class, so you can do something like `digest_run.has_incomplete_subscribers?`.

Reply:
I agree it is a bit of an odd coupling, maybe `mark_complete!` could check whether the `DigestRunSubscriber`s are all marked as complete and if not raise an exception, and then have a method like you suggest `mark_complete_if_able_to` which ignores that exception.

Reply:
I've added a variation on this that encapsulates the whole 'complete' thing in `DigestRun`. I didn't go the exception route as we'd be expecting a lot of the time so it was more control of flow than exceptional. We could add maybe add a validation if you think we should guard it a bit more.